src/App.js

```diff
@@ -1,16 +1,37 @@
 import "./App.css";
 import React, { useState } from "react";
 import Auth from "./pages/Auth";
-import { AuthContext } from "./common/contexts";
 import Home from "./pages/Home";
+import Signup from "./pages/Signup";
+import Login from "./pages/Login";
+import { UserContext, CredContext } from "./common/contexts";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import PrivateRoute from "./common/privateRoute";
 
 function App() {
-  const [auth, setAuth] = useState(false);
+  const [user, setUser] = useState({ email: "", password: "", auth: false });
+  const [creds, setCreds] = useState([]);
```

Comment:
Bad idea to manage the state at the application level.

Reply:
Now, I have created a new Provider component and moved the state inside the provider component. Kindly do let me about this new approach.